src/core/net/dns_client.cpp

```diff
@@ -346,7 +361,8 @@ void Client::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessag
     aMessage.MoveOffset(sizeof(responseHeader));
     offset = aMessage.GetOffset();
 
-    VerifyOrExit((message = FindRelatedQuery(responseHeader, queryMetadata)) != nullptr);
+    VerifyOrExit((message = FindQueryById(responseHeader.GetMessageId())) != nullptr);
+    queryMetadata.ReadFrom(*message);
```

Comment:
'queryMetadata.ReadFrom(*message)' can be moved inside for loop few lines later, just before the following call (where it is needed): `FinalizeDnsTransaction(*message, queryMetadata, &record.GetAddress(), record.GetTtl(), OT_ERROR_NONE);`

Reply:
The `queryMetadata` is used also within `exit` label. Of course we could add `queryMetadata.ReadFrom(*message);` in two places before `FinalizeDnsTransaction`, but that would make sense only, if `queryMetadata.ReadFrom(*message);` was not needed in some of the flow, however i don't see such case i.e. after the Query message is found (line 364), `queryMetadata` has to be always read from the message.

Reply:
Right. My comment was invalid.